test/jdk/java/awt/Frame/Iconify/IconifiedToFront.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, JetBrains s.r.o.. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @summary Verifies that an iconified window is restored with Window.toFront()
+ * @requires os.family == "linux"
+ * @library /test/lib
+ * @run main IconifiedToFront
```

Comment:
Well, the testing noticed for me that this fails because you don't specify 
@key headless

Also the requires os.family == linux surely should not be needed
If this is the desired JDK behaviour it should do the same on all platforms.

Reply:
I'm assuming you meant `@key headful`, right?

The test, unfortunately, fails both on macOS and Windows. I guess the same thing has be done there as well, but I don't have the hardware handy at the moment. Will get back to it later.